src/main/java/roomescape/repository/jpa/ReservationSpecification.java

```diff
@@ -0,0 +1,50 @@
+package roomescape.repository.jpa;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+import java.time.LocalDate;
+import org.springframework.data.jpa.domain.Specification;
+import roomescape.domain.Reservation;
+
+public class ReservationSpecification {
+
+    public static Specification<Reservation> hasMemberId(final Long memberId) {
+        return (Root<Reservation> root, CriteriaQuery<?> query, CriteriaBuilder cb) -> {
+            if (memberId == null) {
+                return cb.conjunction();
+            }
+            return cb.equal(root.get("memberId"), memberId);
+        };
+    }
+
+    public static Specification<Reservation> hasThemeId(final Long themeId) {
+        return (Root<Reservation> root, CriteriaQuery<?> query, CriteriaBuilder cb) -> {
+            if (themeId == null) {
+                return cb.conjunction();
+            }
+            return cb.equal(root.get("themeId"), themeId);
```

Comment:
```suggestion
            return cb.equal(root.get("theme").get("id"), themeId);
```
이부분 동작하지 않는데, 왜그럴까요~?